### PR TITLE
e2e: add first e2e test for cluster logging.

### PR DESCRIPTION
**Pull request by QiaolingTang**
This PR is to add e2e test for cluster logging. @anpingli  @jcantrill @ewolinetz @richm  Could you please help to review it?

Logs:
started: (0/1/1) "[Feature:Logging][Serial] Logging Deploy Logging Via OLM:  should deploy logging successfully when using fluentd as logcollector [Suite:openshift/conformance/serial]"
......

passed: (5m19s) 2019-07-12T03:45:33 "[Feature:Logging][Serial] Logging Deploy Logging Via OLM:  should deploy logging successfully when using fluentd as logcollector [Suite:openshift/conformance/serial]"

Timeline:

Jul 12 03:40:31.320 I ns/openshift-logging clusterserviceversion/clusterlogging.4.1.4-201906271212 requirements not yet checked
Jul 12 03:40:31.357 I ns/openshift-logging clusterserviceversion/clusterlogging.4.1.4-201906271212 one or more requirements couldn't be found
Jul 12 03:40:31.468 I ns/openshift-logging clusterserviceversion/clusterlogging.4.1.4-201906271212 one or more requirements couldn't be found (2 times)
Jul 12 03:40:33.077 I ns/openshift-logging clusterserviceversion/clusterlogging.4.1.4-201906271212 all requirements found, attempting install
......
1 pass, 0 skip (5m19s)

### CODE REVIEW

**Comment by richm** [comment]
I like this very much.  I'm not sure about the go coding, or if it fits in with the origin go test coding conventions, so I'll leave that for the other reviewers.  If origin isn't willing to accept it, then I think we should include it in cluster-logging-operator, or perhaps origin-aggregated-logging.  This is a good foundation for doing more testing in golang rather than our existing origin-aggregated-logging bash "framework".

**Comment by QiaolingTang** [comment]
Thank you, @richm .

**Comment by barleyer** [comment]
IMO,there are some funcs already for us to use like `WaitForNamespacesDeleted`, we dont need to re-invent the wheel,Thanks.

**Comment by QiaolingTang** [comment]
Oh, I'll update my script, thanks @barleyer .

**Comment by anpingli** [comment]
@ironcladlou Could you review this PR.  1) Logging isn't a CVO managed Operators, it isn't deployed by default.  what tags should be used and what tags shouldn't be used for logging cases. 2) To make the PR to be reviewed fastly and correctly?  Could we create our own OWNERS file? How the OWNERS file be defined?

File: test/extended/clusterlogging/deploy_logging.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+var (
+	//CLO resource files to create subscription for cluster-logging-operator
+	CLO = OperatorObjects{
```
Comment (jcantrill) [inline test/extended/clusterlogging/deploy_logging.go]: A more representative test would be to create these resources by way of OLM or what OLM creates.  This is what we are being encouraged to do in our own CI

**Comment by anpingli** [comment]
@smarterclayton,   As a day 2 addon to OCP, could we push extent test PR in the origin repo?  so logging  test cases can be intergrated to openshift-tests and the logging cases can be executed in the build CI or a separate CI job.

**Comment by richm** [comment]
what's holding up this PR?

**Comment by richm** [comment]
@jcantrill is there anything here that can be used for cluster-logging-operator?

**Comment by jcantrill** [comment]
> @jcantrill is there anything here that can be used for cluster-logging-operator?

What does 'can be used' mean?  As in do we have reusable code to deploy CLO?

**Comment by richm** [comment]
> > @jcantrill is there anything here that can be used for cluster-logging-operator?
> 
> What does 'can be used' mean? As in do we have reusable code to deploy CLO?

The golang code to check if things are deployed primarily - I think we're using the olm test script to deploy logging with operators, so we don't need that

**Comment by jcantrill** [comment]
> > > @jcantrill is there anything here that can be used for cluster-logging-operator?
> > 
> > 
> > What does 'can be used' mean? As in do we have reusable code to deploy CLO?
> 
> The golang code to check if things are deployed primarily - I think we're using the olm test script to deploy logging with operators, so we don't need that

Example used for LF
https://github.com/openshift/cluster-logging-operator/blob/master/test/e2e/logforwarding/elasticsearchmanaged/clo_managed_logstore_test.go#L22

**Pull request closed without merging**